### PR TITLE
feat: support darwin arm64

### LINKED ISSUE #8693: Have ARM64 Mac OS X release package
## Feature Request

Opening a feature request kicks off a discussion.

### Proposal:
In the coming releases, have a new row for the new platform/architecture: https://github.com/influxdata/telegraf/releases/tag/v1.17.0

### Current behavior:
Only AMD64 CPU architecture is supported for the Mac OS X Platform.

### Desired behavior:
I can download and easily install/deploy/use the released official binaries, without fuss to compile it.

### Use case: 

To support new&future apple M1 based machines.
Our software needs to be tested on such platforms, to ensure support for the end users.
In our CI we try to monitor all hardware, to early detect issues and help later investigations.
Currently our new hardwares are functioning, and we want to receive telegraf metrics from it.
Our inly option is currently to try to compile it. It is an extra effort, if successful at all.

### PR DESCRIPTION
**Pull request by MyaLongmire**
### Required for all PRs:

- [ ] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #8693 

Add darwin arm64 to the makefile and circle ci

### CODE REVIEW

**Comment by powersj** [comment]
This is awesome! Can you please also update the docs/NIGHTLIES.md file to include downloads for both, please?

I'll go try out the arm64 artifacts from this MP on my M1.

**Comment by powersj** [comment]
The CircleCI artifact looks good. It is not signed, so I do get a prompt about not running it, but I can bypass that. When we do our release we will want to ensure that we sign both the amd64 and arm64 binaries.

**Comment by MyaLongmire** [comment]
> This is awesome! Can you please also update the docs/NIGHTLIES.md file to include downloads for both, please?

Done! :)

> It is not signed, so I do get a prompt about not running it

Is there anything I need to do about that here or just during the release?

**Comment by powersj** [comment]
> Is there anything I need to do about that here or just during the release?

For the release, the only thing here that will need updating is `scripts/mac-signing.sh`. We would need it to sign both files. Right now it is hard coded for `darwin_amd64.tar` only.

**Comment by MyaLongmire** [comment]
> For the release, the only thing here that will need updating is scripts/mac-signing.sh. We would need it to sign both files. Right now it is hard coded for darwin_amd64.tar only.

Now it is a loop that goes over both files.

File: scripts/mac-signing.sh [hunk 1]
```diff
@@ -4,69 +4,76 @@ sudo security import MacCertificate.p12 -k /Library/Keychains/System.keychain -P
...
+    exit 1
+  fi
+
+  # Attach the notarization to the DMG.
+  xcrun stapler staple "$baseName".dmg
+  rm -rf Telegraf.app
+  rm -rf $extractedFolder
+  ls
+
+  echo "${macFiles[$tarFile]} Signed and notarized!"
```
Comment (powersj) [inline scripts/mac-signing.sh:78]: Shouldn't this be `$tarFile`?

Comment (MyaLongmire) [reply scripts/mac-signing.sh:78]: I think it would do the same thing but yours is more straightforward.

File: docs/NIGHTLIES.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ These builds are generated from the master branch each night:
...
+| [amd64.deb](https://dl.influxdata.com/telegraf/nightlies/telegraf_nightly_amd64.deb)   | [aarch64.rpm](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly.aarch64.rpm) | [darwin_amd64.tar.gz](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly_darwin_amd64.tar.gz)       | [darwin_aarm64.tar.gz](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly_darwin_arm64.tar.gz)       | [windows_amd64.zip](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly_windows_amd64.zip) |
 | [arm64.deb](https://dl.influxdata.com/telegraf/nightlies/telegraf_nightly_arm64.deb)   | [armel.rpm](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly.armel.rpm)   | [freebsd_amd64.tar.gz](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly_freebsd_amd64.tar.gz)      | [windows_i386.zip](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly_windows_i386.zip) |
```
Comment (powersj) [inline docs/NIGHTLIES.md:9]: I know this file sucks to edit, but if you go view it in a Markdown generator, you will see you are off by one column. Once this is fixed up I am +1!

Comment (MyaLongmire) [reply docs/NIGHTLIES.md]: This should be fixed!

**Comment by powersj** [comment]
```
/github/workspace/docs/NIGHTLIES.md:17:151 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
/github/workspace/docs/NIGHTLIES.md:22:158 MD047/single-trailing-newline Files should end with a single newline character
```

The file looks better, but in the preview, the `linux_armel.tar.gz` file doesn't have a link.

**Comment by MyaLongmire** [comment]
> The file looks better, but in the preview, the `linux_armel.tar.gz` file doesn't have a link.

The h from https was missing :facepalm:

**Comment by powersj** [comment]
For any reviewers, we are *not* merging this until after 1.21 goes out the door.

File: docs/NIGHTLIES.md [hunk 3]
```diff
@@ -5,15 +5,16 @@ These builds are generated from the master branch each night:
...
-| [s390x.deb](https://dl.influxdata.com/telegraf/nightlies/telegraf_nightly_s390x.deb)   |                 | [linux_i386.tar.gz](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly_linux_i386.tar.gz)         | |
+| [amd64.deb](https://dl.influxdata.com/telegraf/nightlies/telegraf_nightly_amd64.deb)   | [aarch64.rpm](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly.aarch64.rpm) | [darwin_amd64.tar.gz](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly_darwin_amd64.tar.gz)       |  [windows_amd64.zip](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly_windows_amd64.zip) |
+| [arm64.deb](https://dl.influxdata.com/telegraf/nightlies/telegraf_nightly_arm64.deb)   | [armel.rpm](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly.armel.rpm)   | [darwin_aarm64.tar.gz](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly_darwin_arm64.tar.gz)     |  [windows_i386.zip](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly_windows_i386.zip) |
```
Comment (powersj) [inline docs/NIGHTLIES.md:9]:
```suggestion
| [arm64.deb](https://dl.influxdata.com/telegraf/nightlies/telegraf_nightly_arm64.deb)   | [armel.rpm](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly.armel.rpm)   | [darwin_arm64.tar.gz](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly_darwin_arm64.tar.gz)     |  [windows_i386.zip](https://dl.influxdata.com/telegraf/nightlies/telegraf-nightly_windows_i386.zip) |
```

One last change :) extra `a`

Comment (MyaLongmire) [reply docs/NIGHTLIES.md]: Damn, I caught that earlier and thought I fixed it

**Review by powersj — APPROVED** [review]

**Comment by powersj** [comment]
v1.21.0 is out! let's do this!

**Pull request merged**